packages/block-library/src/navigation/style.scss

```diff
@@ -55,6 +57,16 @@
 			&:last-of-type > a {
 				padding-right: 0;
 			}
+
+			// Make submenu icon inherit current text color
+			& > a svg {
+				fill: currentColor;
+			}
+
+			// Add some space between link text & submenu icon
+			& > a > span + span {
```

Comment:
I suggest here using the CSS class added in the markup. Maybe something like the following?

```css
& > a > .wp-block-navigation-link__submenu-icon {
    // ....
}
```

With this, I think we can get rid of adding the extra `<span>` element.

Reply:
FYI, @jeryj asked a similar question before [here](https://github.com/WordPress/gutenberg/pull/19601#discussion_r367614899). I wanted to leave this one to be picked up in the design PR also, so I just added some comments for clarity. Do you think it's worth updating it at this point?

Reply:
On second thought, it should actually be less confusing if I flatten it and just use the dedicated class name :D Updating it now!

Reply:
Just FYI and to give more context here - currently, all the front-end styles for the `navigation-item` block come from the `navigation` block, while the editor ones are inside `navigation-item`. This inconsistency is something that needs to be addressed at some point, but it's way out of this PR's scope.